project/Publish.scala

```diff
@@ -20,17 +20,49 @@ import sbtrelease.ReleasePlugin._
 
 object Publish extends Build {
 
-  override lazy val settings = Seq(
-    credentials += Credentials(Path.userHome / ".ivy2" / ".credentials"),
+  val altReleaseDeploymentRepository = sys.props.get("publish.repository.releases")
+  val altSnapshotDeploymentRepository = sys.props.get("publish.repository.snapshots")
+
+  val nexus = "https://oss.sonatype.org"
+  val defaultReleaseDeploymentRepository = nexus + "/service/local/staging/deploy/maven2"
+  val defaultSnapshotDeploymentRepository = nexus + "/content/repositories/snapshots"
+
+  val releasesDeploymentRepository =
+    "releases" at (altReleaseDeploymentRepository getOrElse defaultReleaseDeploymentRepository)
+  val snapshotsDeploymentRepository =
+    "snapshots" at (altSnapshotDeploymentRepository getOrElse defaultSnapshotDeploymentRepository)
+
+  val altCredentialsLocation = sys.props.get("publish.repository.credentials.file").map(new File(_))
+  val defaultCredentialsLocation = Path.userHome / ".ivy2" / ".credentials"
+  val credentialsLocation = altCredentialsLocation getOrElse defaultCredentialsLocation
+
+  val inlineCredentials = for (
+    realm ← sys.props.get("publish.repository.credentials.realm");
+    host ← sys.props.get("publish.repository.credentials.host");
+    user ← sys.props.get("publish.repository.credentials.user");
+    password ← sys.props.get("publish.repository.credentials.password")
+  ) yield Credentials(realm, host, user, password)
+
+  val resolvedCredentials = inlineCredentials getOrElse Credentials(credentialsLocation)
+
+  println(s"Using $releasesDeploymentRepository for releases")
+  println(s"Using $snapshotsDeploymentRepository for snapshots")
+
+  lazy val creds = Seq(credentials += resolvedCredentials)
+
```

Comment:
- If inline gets precedence now, then why build the credentialsLocation first if it may not even be needed?
- Why all the vals vs lazy val?  publish tasks are rarely used 


```lazy  val inlineCredentials = for (
    realm       <- sys.props.get("publish.repository.credentials.realm")
    host         <- sys.props.get("publish.repository.credentials.host")
    user         <- sys.props.get("publish.repository.credentials.user")
    password <-sys.props.get("publish.repository.credentials.password")
) yield Credentials(realm, host, user, password)```
 
```lazy val resolvedCredentials = inlineCredentials getOrElse {
    val location = sys.props.get("publish.repository.credentials.file")
        .map(new File(_)) getOrElse Path.userHome / ".ivy2" / ".credentials" 
  Credentials(location)
}```

```lazy val creds = Seq(credentials += resolvedCredentials)```